libs/MVS/SceneTexture.cpp

```diff
@@ -151,24 +151,45 @@ struct MeshTexture {
 		typedef TRasterMesh<RasterMesh> Base;
 		FaceMap& faceMap;
 		FIndex idxFace;
+		cv::Mat mask;
+		bool validFace;
+		std::vector<ImageRef> prevPixels;
+
 		RasterMesh(const Mesh::VertexArr& _vertices, const Camera& _camera, DepthMap& _depthMap, FaceMap& _faceMap)
-			: Base(_vertices, _camera, _depthMap), faceMap(_faceMap) {}
+			: Base(_vertices, _camera, _depthMap), faceMap(_faceMap), validFace(true) {}
 		void Clear() {
 			Base::Clear();
 			faceMap.memset((uint8_t)NO_ID);
 		}
+		// TODO: EDIT THIS FUNCTION TO FILTER
 		void Raster(const ImageRef& pt, const Point3f& bary) {
 			const Point3f pbary(PerspectiveCorrectBarycentricCoordinates(bary));
 			const Depth z(ComputeDepth(pbary));
 			ASSERT(z > Depth(0));
 			Depth& depth = depthMap(pt);
+			if (!validFace) {
+				depth = -1;
```

Comment:
as we discussed and tried to explain also on slack, pls do not touch the depth, depth should still record every single face, valid or not so that it do not allow faces in the back to be projected on the image; lets discuss this farther if not clear